programs/openbook-v2/src/instructions/close_market.rs

```diff
@@ -10,6 +11,18 @@ pub fn close_market(ctx: Context<CloseMarket>) -> Result<()> {
         ctx.accounts.close_market_admin.key() == close_admin,
         OpenBookError::InvalidCloseMarketAdmin
     );
+    // check market is expired
+    require!(
+        market.time_expiry == -1 || market.time_expiry < Clock::get()?.unix_timestamp,
```

Comment:
why is there a `set_market_expired` that sets `time_expiry = -1` for expired markets if this function can also be called when the market is expired?

Reply:
If you want to expire a market that initially has time_expiry = 0 that's the way you do it

Reply:
oh i see. I don't know if I really like the idea that a market can be arbitrarily closed (even if it is permisioned). at least, i think the `set_market_expired`, should change the expiration time not to `-1`, but to some point in the near future, requiring at lest 1 day/a few hours until the market is actually expired so users cannot found that suddenly & without notice the market is not longer under operation